cache/cache.go

```diff
@@ -0,0 +1,154 @@
+package cache
+
+import (
+	"io"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/karimra/gnmic/formatters"
+	"github.com/karimra/gnmic/outputs"
+	"github.com/karimra/gnmic/utils"
+	gnmiCache "github.com/openconfig/gnmi/cache"
+	"github.com/openconfig/gnmi/ctree"
+	"github.com/openconfig/gnmi/proto/gnmi"
+)
+
+const (
+	loggingPrefix  = "[cache] "
+	defaultTimeout = 10 * time.Second
+	defaultFlush   = 5 * time.Second
+)
+
+type GnmiCacheConfig struct {
+	Enabled    bool          `mapstructure:"enabled,omitempty"`
+	Expiration time.Duration `mapstructure:"expiration,omitempty"`
+	Timeout    time.Duration `mapstructure:"timeout,omitempty"`
+	FlushTimer time.Duration `mapstructure:"flush-timer,omitempty"`
```

Comment:
Should this be an output configuration option instead of a cache option?  Flushing the cache is technically controlled by the output plugin